core/proposal/genesis.go

```diff
@@ -253,3 +258,32 @@ func LoadFromFile(file string) (*Genesis, error) {
 	}
 	return &gen, nil
 }
+
+
+/* save genesis file to file system */
+func (gen *Genesis) Save(workingDir string) string {
+
+	wDir := " "
```

Comment:
I think better to change it like : 
```
if workingDir != "" {
		workingDir = "/tmp/chain/"
	}
```

In this case you check the parameter to not be empty and give value if it is empty. Less code is blessed...